apps/builder/app/builder/features/breakpoints/find-initial-width.ts

```diff
@@ -11,12 +12,24 @@ export const findInitialWidth = (
   selectedBreakpoint: Breakpoint,
   workspaceWidth: number
 ) => {
-  // When on a base breakpoint, we want to show maximum width canvas
-  if (
-    selectedBreakpoint.minWidth === undefined &&
-    selectedBreakpoint.maxWidth === undefined
-  ) {
-    return workspaceWidth;
+  if (isBaseBreakpoint(selectedBreakpoint)) {
+    // Finding the canvas width when user selects base breakpoint is a bit more complicated.
+    // We want to find the lowest possible size that is bigger than all max breakpoints and smaller than all min breakpoints.
```

Comment:
Wrong comment, the algorithm below is right only if groupedBreakpoints returns non-intersected Set of min max breakpoints that's not true. Example.
<img width="256" alt="image" src="https://user-images.githubusercontent.com/5077042/233556495-6500a783-2248-4a32-9e9b-1bad03652247.png">
 
So here it returns just `return Math.max(next.maxWidth + 1, defaultWidth);`
`groupBreakpoints` have no logic that resulting groups must not intersect

Reply:
yes, we need to decide what to do about intersecting min and max breakpoints, what happens if we have a 300px canvas and there is min and max breakpoint that applies at that width? We would have 2 lines one is on the right, one is on the left.

maybe that's alright, but user shouldn't want to have this in practice because there is no point in defining a breakpoint that has an intersection, since you are trying to address unique resolutions

Reply:
added a comment regarding intersecting breakpoints for now, not sure we should prevent user from ever having an intersection, because reasons for having an intersection could be complex e.g. copy/paste and user should decide how they define them